source/plugins/go/src/log_loss_monitoring.go

```diff
@@ -0,0 +1,303 @@
+				for {
+					select {
+					case deleted_identifier := <-deleted_containers_response:
+						m_bytes_logged_storage.delete(deleted_identifier)
+					default:
+						break read_from_chan
+					}
+				}
+			}
+		}
+		log_bytes := len(*logTime) + len(" stdout f ") + len(*logEntry) + 1 // (an extra byte for the trailing \n in the source log file)
+
+		// double check that the atomic read/write fixes any concurency issues? (it really should)
+		atomic.AddInt64(log_count_index, int64(log_bytes))
+	}
+}
+
+func write_telemetry(ticker <-chan time.Time) {
+	// putting this code in a sub-function so that it can use defer
+	inner_func := func() {
+		identifiers, values := m_bytes_logged_storage.export_values()
+
+		read_disk_mut.Lock()
+		defer read_disk_mut.Unlock()
+
+		var total_bytes_logged int64 = 0
+		var total_bytes_on_disk int64 = 0
+
+		for index, container_identifier := range identifiers {
+			if container_identifier == "" {
+				continue
+			}
+			logs_counted := values[index]
+
+			record, container_seen_on_disk := FW_records[container_identifier]
+			rotated_bytes := record.rotated_bytes
+			unrotated_bytes := record.unrotated_bytes
+
+			if !container_seen_on_disk {
+				// this can happen for perfectly normal reasons (like the container is waiting to be garbage collected)
+				continue
+			}
+
+			total_bytes_logged += logs_counted
+			total_bytes_on_disk += rotated_bytes + unrotated_bytes
+			log_loss_logger.Printf(`{"namespace_pod_container": "%s", "bytes_at_log_file": %d, "bytes_at_output_plugin": %d}`, container_identifier, rotated_bytes+unrotated_bytes, logs_counted)
+		}
+		// these variables live in telemetry.go
+		LogsLostInFluentBit = total_bytes_on_disk - total_bytes_logged
+		LogsWrittenToDisk = total_bytes_on_disk
+	}
+
+	for range ticker {
+		inner_func()
+	}
+}
+
+func track_log_rotations(ticker <-chan time.Time, watch_dir string) {
+	inner_func := func() {
+		read_disk_mut.Lock()
+		defer read_disk_mut.Unlock()
+
+		// this needs a way to return an error
+		all_files_and_sizes := GetSizeOfAllFilesInDir(watch_dir)
+		containers_seen_this_iter := make(map[string]bool)
+
+		for filepath, file_size := range all_files_and_sizes {
+			// we don't care about compressed log files
+			if strings.HasSuffix(filepath, ".gz") {
+				continue
+			}
+
+			filepath_parts := strings.Split(filepath, "/") // TODO: this will be different for windows
+
+			//TODO: need some safety here
+			if len(filepath_parts) != 3 {
+				Log("illegal file found in pod log dir: %v", filepath)
+				continue
+			}
+			pod_folder_parts := strings.Split(filepath_parts[0], "_") // this is safe because kubernetes objects can't have _ in their name
+			namespace_folder := pod_folder_parts[0]
+			pod_folder := pod_folder_parts[1]
+			container_folder := filepath_parts[1]
+			log_file_name := filepath_parts[2]
+			container_identifier := namespace_folder + "_" + pod_folder + "_" + container_folder // should be of the format namespace_podname-garbage_containername
```

Comment:
format is namespace_<podname>_containerid.  podname is podname-<hashvalue>

Reply:
The format is pod-name_pod-uuid/container-name/log-file.log since we're looking at the pods log folder instead of the containers log folder